src/main/java/com/zepben/cimbend/grpc/GrpcChannelFactory.kt

```diff
@@ -25,20 +25,19 @@ object GrpcChannelFactory {
         val channelBuilder = NettyChannelBuilder.forAddress(config.host, config.port)
 
         if (config.enableTls) {
-            if (!config.privateKeyFilePath.isNullOrBlank() && !config.certChainFilePath.isNullOrBlank()) {
-                val sslContextBuilder = GrpcSslContexts.forClient()
+            val sslContextBuilder = GrpcSslContexts.forClient()
 
-                sslContextBuilder.keyManager(File(config.certChainFilePath), File(config.privateKeyFilePath))
+            if (!config.trustCertPath.isNullOrBlank())
+                sslContextBuilder.trustManager(File(config.trustCertPath))
 
-                if (!config.trustCertCollectionFilePath.isNullOrBlank()) {
-                    sslContextBuilder.trustManager(File(config.trustCertCollectionFilePath))
-                }
-
-                val sslContext = GrpcSslContexts.configure(sslContextBuilder).build()
-                channelBuilder.sslContext(sslContext)
-            } else {
-               throw BadConfigException("If TLS is enabled you must specify at least a key and cert")
+            if (!config.authCertPath.isNullOrBlank()) {
+                if (!config.authKeyPath.isNullOrBlank())
+                    sslContextBuilder.keyManager(File(config.authCertPath), File(config.authKeyPath))
+                else
+                    throw BadConfigException("If TLS auth is enabled you must specify a key and cert")
             }
+
+            channelBuilder.sslContext(GrpcSslContexts.configure(sslContextBuilder).build())
         } else
             channelBuilder.usePlaintext()
```

Comment:
What happens if this runs with a default `ConnectionConfig` (i.e, enableTls=true with no key/cert/ca)? Does it fail with a useful error?

I set enableTls = false by default so that if you just accepted the defaults it wouldn't throw any errors.

Reply:
As it is a client connection not a server it does not need a certificate to use TLS. If you do not provide a CA it should use hte system default, which I assume means it will check you local certificate cache, which you can install the CA into.

Reply:
ah right, i think in that case we should still default to false because no one is going to have the CA installed into the jvms default cert chain, so it would fail.